packages/core/lib/package.js

```diff
@@ -1,351 +1,214 @@
 const expect = require("@truffle/expect");
 const TruffleError = require("@truffle/error");
-const Networks = require("./networks");
-const EthPM = require("ethpm");
-const EthPMRegistry = require("ethpm-registry");
-const Web3 = require("web3");
-const { createInterfaceAdapter } = require("@truffle/interface-adapter");
-const async = require("async");
+const { EthPM } = require("ethpm");
+const { Erc1319URI } = require("ethpm/utils/uri");
+const { parseTruffleArtifacts } = require("ethpm/utils/truffle");
 const path = require("path");
 const fs = require("fs");
-const OS = require("os");
 
 const Package = {
-  install: async function(options, callback) {
-    expect.options(options, ["working_directory", "ethpm"]);
-
-    expect.options(options.ethpm, ["registry", "ipfs_host"]);
-
-    expect.one(options.ethpm, ["provider", "install_provider_uri"]);
-
-    // ipfs_port and ipfs_protocol are optinal.
+  install: async options => {
+    expect.options(options.ethpm, ["ipfs_host", "ipfs_port", "ipfs_protocol"]);
+    expect.options(options, ["working_directory", "ethpm_uri", "logger"]);
 
+    // Create a web3 instance connected to a blockchain
+    if (!options.provider && !options.ethpm.install_provider_uri) {
+      throw new TruffleError("No provider or install_provider_uri found.");
+    }
     const provider =
-      options.ethpm.provider ||
-      new Web3.providers.HttpProvider(options.ethpm.install_provider_uri, {
-        keepAlive: false
-      });
-    let host = options.ethpm.ipfs_host;
-
-    if (host instanceof EthPM.hosts.IPFS === false) {
-      host = new EthPM.hosts.IPFSWithLocalReader(
-        options.ethpm.ipfs_host,
-        options.ethpm.ipfs_port,
-        options.ethpm.ipfs_protocol
+      options.provider ||
+      new Web3.providers.HttpProvider(options.ethpm.install_provider_uri);
+
+    // Parse ethpm uri
+    const ethpmURI = new Erc1319URI(options.ethpm_uri);
```

Comment:
btw if you get sick of typing `options.ethpm_uri`, you can do this kinda business:

```javascript
const { ethpm_uri } = options
```